src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php

```diff
@@ -360,4 +360,33 @@ protected function getServiceEdges(ContainerBuilder $container, string $serviceI
             return [];
         }
     }
+
+    /**
+     * Filters the given collection of routes by the specified HTTP method.
+     * This function iterates through the provided routes and removes those
+     * whose allowed methods do not include the specified HTTP method. If
+     * no method is provided (i.e., it is `null`), the original collection
+     * of routes is returned unchanged.
+     *
+     * @param RouteCollection $routes the collection of routes to be filtered
+     * @param string|null     $method The HTTP method to filter routes by (e.g., 'GET', 'POST'). If `null`, no filtering is applied.
+     *
+     * @return RouteCollection the filtered collection of routes
+     */
+    private function filterRoutesByHttpMethod(RouteCollection $routes, ?string $method): RouteCollection
```

Comment:
What about let the call filter only if there is a method?
```suggestion
    private function filterRoutesByHttpMethod(RouteCollection $routes, string $method): RouteCollection
```
eg in the caller:
```php
$object instanceof RouteCollection => $options['method'] ? $this->describeRouteCollection($this->filterRoutesByHttpMethod($object, $options['method']), $options) : $this->describeRouteCollection($object, $options),
```

Reply:
I will definitely make that change, thanks for the suggestions.